tests/jballerina-unit-test/src/test/resources/test-src/expressions/varref/record-variable-reference-semantics-negative.bal

```diff
@@ -198,3 +198,23 @@ function testFieldAndIndexBasedVarRefs() returns [anydata, anydata] {
     {name: m["var1"], yearAndAge: [m["var2"], _]} = ch3;
     return [m["var1"], m["var2"]];
 }
+
+type Employee record {
+    string name;
+    int id;
+    int age?;
+};
+
+function testOptionalFieldsInRecordBindingPattern(){
+   Employee e = {name: "Jo", id: 1234};
+
+   string eName;
+   int eId;
+   int eAge;
+
+   {name: eName, id: eId, age: eAge} = e;
+
+   var {name: name, id: id, age: age} = e;
+
+   var {name: nameNew, id: idNew, age: ageNew} = e;
```

Comment:
Is this the correct file for these tests? Doesn't this file contain only destructuring assignments?